agent/agent.go

```diff
@@ -265,20 +277,102 @@ func (a *Agent) loop(ctx context.Context) error {
 	if a.opts.MaxAgentSleepInterval != 0 {
 		maxAgentSleepInterval = a.opts.MaxAgentSleepInterval
 	}
-	agentSleepInterval := minAgentSleepInterval
+	taskCtx, cancel := context.WithCancel(ctx)
 
-	var jitteredSleep time.Duration
-	tskCtx, cancel := context.WithCancel(ctx)
-	defer cancel()
+	lc := &loop{
+		tc:                    &taskContext{},
+		needPostGroup:         false,
+		taskCtx:               taskCtx,
+		cancel:                cancel,
+		timer:                 time.NewTimer(0),
+		minAgentSleepInterval: minAgentSleepInterval,
+		maxAgentSleepInterval: maxAgentSleepInterval,
+		agentSleepInterval:    minAgentSleepInterval,
+	}
+
+	return lc
+}
+
+func (a *Agent) processNextTask(ctx context.Context, l *loop, nextTask *apimodels.NextTaskResponse) (shouldContinue bool, shouldExit bool) {
+	if nextTask.ShouldExit {
+		grip.Notice("Next task response indicates agent should exit.")
+		shouldExit = true
+		return
```

Comment:
It's considered better practice to explicitly return the values rather than rely on setting the output variable names. This is because it becomes difficult to trace what's true vs. false if you have to track the flow of variable logic. (Same with the other places in this function.)